packages/nx/src/plugins/js/utils/register.ts

```diff
@@ -35,6 +35,59 @@ export const registerTsProject = (
   };
 };
 
+export function getSwcTranspiler(
+  compilerOptions: CompilerOptions
+): (...args: unknown[]) => unknown {
+  // Some versions of @swc-node/register do not return a cleanup function, so we explicitly check for it here
+  type ISwcRegister = typeof import('@swc-node/register/register')['register'];
+  type ISwcRegisterVoid = (...args: Parameters<ISwcRegister>) => void;
+
+  // These are requires to prevent it from registering when it shouldn't
+  const register = require('@swc-node/register/register').register as
+    | ISwcRegister
+    | ISwcRegisterVoid;
```

Comment:
Add explicit type-checking to account for versions of `@swc-node/register` that do not return cleanup functions.

Reply:
I'm removing some of this, since we are removing the strict portion it doesn't end up helping and looks a bit confusing

Reply:
Makes sense without strict null checks 👍